### PR TITLE
Some fixes

### LINKED ISSUE #3446: Trailblazer's Torch seems to be dealing damage instead of the equipped creature
Current snapshot on Win; I'm attacking with a lifelink creature with Trailblazer's Torch (CLB), but I'm not gaining life from the damage dealt when being blocked. Presumably it's incorrectly dealing the damage itself, instead of having the equipped creature deal the damage.

### LINKED ISSUE #3448: Sovereigns of Lost Alara (ARB) remembers previous creatures as candidates for the enchantment
Current snapshot, Windows. When a creature triggers Sovereigns of Lost Alara, it becomes "remembered". Then on subsequent triggers, you get to choose which of the remembered creatures gets the aura.

For example, I attacked with a creature and got the effect, searching out an aura for it. I then later attacked with Sovereigns of Lost Alara itself, and I was allowed to attach the aura to the creature that had attacked alone on a previous turn.

I'm not sure why it's implemented as a "remembered" effect; it should be able to be forgotten as soon as it resolves.

### LINKED ISSUE #3457: Preston the Vanisher copies itself when entering from the graveyard
**Preston Rules Text**
Whenever another nontoken creature enters the battlefield under your control, if it wasn't cast, create a token that's a copy of that creature, except it's a 0/1 white Illusion.

**Describe the bug**
Preston the Vanisher triggers its own effect when entering the battlefield by any means other than casting. It doesn't disturb the game that much because of the legend rule, but it does make a difference.

**Expected behavior**
I'm pretty sure the effect should only trigger when Preston is already on the field and especially not for the same card.

### LINKED ISSUE #3459: Esquire of the King (LTR) has flying for no reason
Current snapshot, Windows. Esquire of the King _doesn't_ have flying, and the card text in Forge doesn't say it does, except suddenly when it's on the battlefield it does!

### PR DESCRIPTION
**Pull request by tool4ever**
Closes #3448
Closes #3446
Closes #3459
Closes #3457

### CODE REVIEW

**Review by Agetian — APPROVED** [review]

File: forge-gui/res/cardsfolder/s/skyshroud_ambush.txt [hunk 1]
```diff
@@ -2,9 +2,9 @@ Name:Skyshroud Ambush
...
-SVar:DBFight:DB$ Fight | Defined$ ParentTarget | ValidTgts$ Creature.YouDontCtrl | TgtPrompt$ Choose target creature you don't control | RememberObjects$ ThisTargetedCard | SubAbility$ DBEffect | StackDescription$ fights target creature you don't control [{c:ThisTargetedCard}]. When the creature you control wins the fight, draw a card.
-SVar:DBEffect:DB$ Effect | Triggers$ TrigWin | RememberObjects$ Remembered | ImprintCards$ Imprinted | Duration$ UntilStateBasedActionChecked | SubAbility$ DBCleanup
+SVar:DBFight:DB$ Fight | Defined$ ParentTarget | ValidTgts$ Creature.YouDontCtrl | TgtPrompt$ Choose target creature you don't control | SubAbility$ DBEffect | StackDescription$ fights target creature you don't control [{c:ThisTargetedCard}]. When the creature you control wins the fight, draw a card.
+SVar:DBEffect:DB$ Effect | Triggers$ TrigWin | RememberObjects$ ParentTarget | ImprintCards$ Imprinted | Duration$ UntilStateBasedActionChecked | SubAbility$ DBCleanup
```
Comment (Hanmac) [inline forge-gui/res/cardsfolder/s/skyshroud_ambush.txt:6]: How to you restrict to Creature you don't control?

File: forge-gui/res/cardsfolder/v/viviens_invocation.txt [hunk 2]
```diff
@@ -2,7 +2,7 @@ Name:Vivien's Invocation
 ManaCost:5 G G
 Types:Sorcery
 A:SP$ Dig | Cost$ 5 G G | DigNum$ 7 | ChangeNum$ 1 | ChangeValid$ Creature | Optional$ True | RestRandomOrder$ True | DestinationZone$ Battlefield | ForceRevealToController$ True | SpellDescription$ Look at the top seven cards of your library. You may put a creature card from among them onto the battlefield. Put the rest on the bottom of your library in a random order. When a creature is put onto the battlefield this way, it deals damage equal to its power to target creature an opponent controls.
-T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Creature | ValidCause$ Card.Self | Execute$ DBDealDamage | Secondary$ True | TriggerDescription$ When a creature is put onto the battlefield this way, it deals damage equals to its power to target creature an opponent controls.
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Creature | ValidCause$ Spell.Self | Execute$ DBDealDamage | Secondary$ True | TriggerDescription$ When a creature is put onto the battlefield this way, it deals damage equals to its power to target creature an opponent controls.
```
Comment (Hanmac) [inline forge-gui/res/cardsfolder/v/viviens_invocation.txt:5]: Should be a Immediate Trigger spawning from the Effect above

File: forge-gui/res/cardsfolder/s/skyshroud_ambush.txt (hunk 1 above)
Comment (tool4ever) [reply forge-gui/res/cardsfolder/s/skyshroud_ambush.txt:6]: Hmn?
ParentTarget only uses the first targeting SA it finds

File: forge-gui/res/cardsfolder/v/viviens_invocation.txt (hunk 2 above)
Comment (tool4ever) [reply forge-gui/res/cardsfolder/v/viviens_invocation.txt:5]: thanks

**Review by Agetian — DISMISSED** [review]

**Review by kevlahnota — APPROVED** [review]

**Pull request merged**